CMakeLists.txt

```
@@ -0,0 +1,158 @@
+cmake_minimum_required(VERSION 3.0)
```

```suggestion
cmake_minimum_required(VERSION 3.10)
```

So that we can build with CMake >= 4.0

https://cmake.org/cmake/help/latest/release/4.0.html#deprecated-and-removed-features